src/recommendationservice/recommendation_server.py

```diff
@@ -41,6 +49,8 @@ def ListRecommendations(self, request, context):
         span = trace.get_current_span()
         span.set_attribute("app.products_recommended.count", len(prod_list))
         logger.info("[Recv ListRecommendations] product_ids={}".format(prod_list))
+        recommendations_counter.add(len(prod_list), {'recommendation.type': 'catalog'})
```

Comment:
Totally LGTM, one nit not block, should the metric name be more consistent with span attribute? Since attribute is `app.products_recommended.count`, the metric name may be `product_recommended_count`? 

Another question, should manually created metrics start with common prefix? such as `recommandation_`?

Reply:
Interesting questions. I agree we should have consistent naming for metrics. What about “app_” prefix for all manual metrics in the demo app? Then they would be easy to find throughout the app.

We could also create an “inventory” document for manual metrics, similar to this one: 
https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/manual_span_attributes.md

I’m not sure about the best practice to include the service name in the metric name.

Reply:
> 



> Interesting questions. I agree we should have consistent naming for metrics. What about “app_” prefix for all manual metrics in the demo app? Then they would be easy to find throughout the app.
> 
> We could also create an “inventory” document for manual metrics, similar to this one:  https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/manual_span_attributes.md
> 
> I’m not sure about the best practice to include the service name in the metric name.

@reyang do you have any suggestions?

Reply:
Agreed in the SIG meeting that let's go with “app_” prefix.

Reply:
> @reyang do you have any suggestions?

+1 on having "app_" prefix to encourage consistency across the demo.

In reality, production services are very likely to have existing metrics and downstream integration (e.g. alert rules, auto-scaling rules) so it'll take time and efforts for them to change the names (whether it is the name of the metrics stream or individual tags/labels), I _guess_ many of them might just decide to keep using existing names (especially considering the continuity of business/operational metrics).

Reply:
Counter name updated accordingly.